homeassistant/components/statistics/sensor.py

```diff
@@ -71,9 +72,15 @@
     sampling_size = config.get(CONF_SAMPLING_SIZE)
     max_age = config.get(CONF_MAX_AGE, None)
     precision = config.get(CONF_PRECISION)
+    debounced_updater = Debouncer(hass, _LOGGER, cooldown=0.01, immediate=False)
```

Comment:
Why `immediate` not set to `True` ?

Reply:
because if 3 samples are received, we want to make the update (data process) with all samples, so at the end of the grouped calls

Reply:
(If I understood correctly the code, both in debouncer and in statistics)

Update call does the purging, process of all attributes, and state definition. If called in the first sample, the data processed is not correct, as does not mirror all the received samples, right?

Reply:
I just realized that I've updated our debouncer like a throttle. It will ensure a maximum of calling a command once per `<cooldown>` period, instead of after `<cooldown>` period, with each call resetting the `<cooldown>`.

Reply:
I also realized that it isn't 100% perfect, so added some fixes in #32161